app/views/meetings/documents/agenda.js

```diff
@@ -445,13 +464,84 @@ export default ["$scope", "$route", "$http", '$q', '$interval', 'conferenceServi
             _ctrl.types[0].loaded = false;
             _ctrl.currentTab = undefined;
 
-            $location.search({datetime:_ctrl.scheduleDate||undefined});
-            
+            $location.search({ all, datetime: _ctrl.scheduleDate||undefined});
+             
             $timeout(function(){load();_ctrl.currentTab = tab;}, 100)
 
         }
 
         function resData(res) {
             return res.data;
         }
+
+        function getTimezone() {
+          if(!_ctrl.all) return _ctrl.conferenceTimezone
+
+
+          return  localStorage.getItem('timezone') || Intl.DateTimeFormat().resolvedOptions().timeZone
+        }
+        _ctrl.getTimezone = getTimezone
+
+        function getTimezones() {
+          const browserTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone
+
+          if(_ctrl.conferenceTimezone === browserTimezone) {
+            if(localStorage.getItem('timezone') !== _ctrl.conferenceTimezone){
+              localStorage.setItem('timezone', _ctrl.conferenceTimezone)
+              $timeout(function(){load()}, 100)
+            }
+            return [] 
+          }
+
+          if(getTimezone() === _ctrl.conferenceTimezone) return [browserTimezone]
+
+          return [_ctrl.conferenceTimezone]
+        }
+        _ctrl.getTimezones = getTimezones
+
+        function timezoneChanged(zone) {
+          const tz = localStorage.getItem('timezone')
+
+          const selectedTz = zone || tz || Intl.DateTimeFormat().resolvedOptions().timeZone
+
+          localStorage.setItem('timezone', selectedTz )
+
+          $timeout(function(){load()}, 100)
```

Comment:
do you need 100ms? if not can you $applyAsync() instead?
```suggestion
          $scope.$applyAsync(()=>load())
```